trail/trail.go

```diff
@@ -82,37 +84,39 @@ func ToGRPC(err error) error {
 	if err == nil {
 		return nil
 	}
+
+	if errors.Is(err, io.EOF) {
+		return err
+	}
+
 	// If err is already a gRPC error, don't modify it.
 	if _, ok := status.FromError(err); ok {
 		return err
 	}
 
 	userMessage := trace.UserMessage(err)
-	if trace.IsNotFound(err) {
+	err = trace.Unwrap(err)
+	switch err.(type) {
```

Comment:
I'm not sure if we should check the error type instead of calling the various `trace.Is` methods. Some of them return true for other errors as well, i.e: https://github.com/gravitational/trace/blob/master/errors.go#L79.

Perhaps an empty switch a la the following would be a better way to go?

```go
switch {
case trace.IsAlreadyExists(err):
         return grpc.Errorf(codes.AlreadyExists, userMessage)
case trace.IsAccessDenied(err):
         return grpc.Errorf(codes.PermissionDenied, userMessage)
......
}
```

Reply:
good point, I only checked a few implementations and assumed they all follow the same pattern